### ISSUE TITLE
return_linter() should ignore some (all?) terminal repeat{} statements

### ISSUE THREAD

**Issue by MichaelChirico** [jimhester/lintr#3044]
```r
foo <- function() {
  repeat {
    if (TRUE) return(1)
  }
}
```

Fails under `explicit` return rule:

```
# <text>:2:3: warning: [return_linter] All functions must have an explicit return().
#   repeat {
#   ^~~~~~
```

(the same applies if there's an `else` to the `if`)

Need to think more carefully about what exactly to enforce here. The idea is that `repeat {}` can't exit without `break`, so adding `return(invisible())` or something like that after the loop to satisfy the linter is misleading.

It seems like `repeat{}` doesn't have any way of being the terminal statement though:

```r
lintr::lint(linters = lintr::return_linter(return_style="explicit"), text="foo <- function() {
  repeat {
    return(1)
  }
}")
# <text>:2:3: warning: [return_linter] All functions must have an explicit return().
#   repeat {
#   ^~~~~~
```

I think the best we can do (?) is require _any_ call to `return()` _or_ `break` under `repeat {}`.
